lib/server.js

```diff
@@ -753,17 +753,14 @@ Server.prototype._handle = function _handle(req, res) {
     // run pre() handlers first before routing and running
     if (self.before.length > 0) {
         self._run(req, res, null, self.before, function (err) {
-            // check for return false here - like with the regular handlers,
-            // if false is returned we already sent a response and should stop
-            // processing.
-            if (err === false) {
-                self._finishReqResCycle(req, res);
+            // Like with regular handlers, if we are provided an error, we
+            // should abort the middleware chain and fire after events.
+            if (err === false || err instanceof Error) {
+                self._finishReqResCycle(req, res, null, err);
                 return;
             }
 
-            if (!err) {
```

Comment:
How come this got dropped? I think if you remove this there is the edge case where the "err" object is not false or a function?